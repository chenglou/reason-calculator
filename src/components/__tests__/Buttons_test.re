open Jest;
open Expect;

let _ =

describe "Buttons" (fun _ => {
  test "renders" (fun _ => {
    let dispatch = fun _ => ();
    let tree = Buttons.createElement ::dispatch children::[] ()
      |> ReactShallowRenderer.renderWithRenderer;

    expect tree |> toMatchSnapshot;
  });
});
