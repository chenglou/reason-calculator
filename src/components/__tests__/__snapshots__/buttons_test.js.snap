// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders 1`] = `
<div>
  <ButtonGroup
    reasonProps={
      Array [
        Array [
          <GreenButton
            reasonProps={
                  Array [
                        1,
                        "",
                        [Function],
                      ]
            }
      />,
          Array [
            <GreenButton
              reasonProps={
                      Array [
                              7,
                              "",
                              [Function],
                            ]
              }
      />,
            Array [
              <GreenButton
                reasonProps={
                          Array [
                                    8,
                                    "",
                                    [Function],
                                  ]
                }
      />,
              Array [
                <OperationButton
                  reasonProps={
                              Array [
                                          2,
                                          "",
                                          [Function],
                                        ]
                  }
      />,
                0,
              ],
            ],
          ],
        ],
      ]
    }
  >
    <GreenButton
      reasonProps={
        Array [
          1,
          "",
          [Function],
        ]
      }
    />
    <GreenButton
      reasonProps={
        Array [
          7,
          "",
          [Function],
        ]
      }
    />
    <GreenButton
      reasonProps={
        Array [
          8,
          "",
          [Function],
        ]
      }
    />
    <OperationButton
      reasonProps={
        Array [
          2,
          "",
          [Function],
        ]
      }
    />
  </ButtonGroup>
  <ButtonGroup
    reasonProps={
      Array [
        Array [
          <ValueButton
            reasonProps={
                  Array [
                        [Function],
                        0,
                        "7",
                      ]
            }
      />,
          Array [
            <ValueButton
              reasonProps={
                      Array [
                              [Function],
                              0,
                              "8",
                            ]
              }
      />,
            Array [
              <ValueButton
                reasonProps={
                          Array [
                                    [Function],
                                    0,
                                    "9",
                                  ]
                }
      />,
              Array [
                <OperationButton
                  reasonProps={
                              Array [
                                          5,
                                          "",
                                          [Function],
                                        ]
                  }
      />,
                0,
              ],
            ],
          ],
        ],
      ]
    }
  >
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "7",
        ]
      }
    />
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "8",
        ]
      }
    />
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "9",
        ]
      }
    />
    <OperationButton
      reasonProps={
        Array [
          5,
          "",
          [Function],
        ]
      }
    />
  </ButtonGroup>
  <ButtonGroup
    reasonProps={
      Array [
        Array [
          <ValueButton
            reasonProps={
                  Array [
                        [Function],
                        0,
                        "4",
                      ]
            }
      />,
          Array [
            <ValueButton
              reasonProps={
                      Array [
                              [Function],
                              0,
                              "5",
                            ]
              }
      />,
            Array [
              <ValueButton
                reasonProps={
                          Array [
                                    [Function],
                                    0,
                                    "6",
                                  ]
                }
      />,
              Array [
                <OperationButton
                  reasonProps={
                              Array [
                                          6,
                                          "",
                                          [Function],
                                        ]
                  }
      />,
                0,
              ],
            ],
          ],
        ],
      ]
    }
  >
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "4",
        ]
      }
    />
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "5",
        ]
      }
    />
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "6",
        ]
      }
    />
    <OperationButton
      reasonProps={
        Array [
          6,
          "",
          [Function],
        ]
      }
    />
  </ButtonGroup>
  <ButtonGroup
    reasonProps={
      Array [
        Array [
          <ValueButton
            reasonProps={
                  Array [
                        [Function],
                        0,
                        "1",
                      ]
            }
      />,
          Array [
            <ValueButton
              reasonProps={
                      Array [
                              [Function],
                              0,
                              "2",
                            ]
              }
      />,
            Array [
              <ValueButton
                reasonProps={
                          Array [
                                    [Function],
                                    0,
                                    "3",
                                  ]
                }
      />,
              Array [
                <OperationButton
                  reasonProps={
                              Array [
                                          0,
                                          "",
                                          [Function],
                                        ]
                  }
      />,
                0,
              ],
            ],
          ],
        ],
      ]
    }
  >
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "1",
        ]
      }
    />
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "2",
        ]
      }
    />
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          "3",
        ]
      }
    />
    <OperationButton
      reasonProps={
        Array [
          0,
          "",
          [Function],
        ]
      }
    />
  </ButtonGroup>
  <ButtonGroup
    reasonProps={
      Array [
        Array [
          <ValueButton
            reasonProps={
                  Array [
                        [Function],
                        1,
                        "0",
                      ]
            }
      />,
          Array [
            <ValueButton
              reasonProps={
                      Array [
                              [Function],
                              0,
                              ".",
                            ]
              }
      />,
            Array [
              <OperationButton
                reasonProps={
                          Array [
                                    3,
                                    "",
                                    [Function],
                                  ]
                }
      />,
              0,
            ],
          ],
        ],
      ]
    }
  >
    <ValueButton
      reasonProps={
        Array [
          [Function],
          1,
          "0",
        ]
      }
    />
    <ValueButton
      reasonProps={
        Array [
          [Function],
          0,
          ".",
        ]
      }
    />
    <OperationButton
      reasonProps={
        Array [
          3,
          "",
          [Function],
        ]
      }
    />
  </ButtonGroup>
</div>
`;
